File: lib/core/utils/fees.dart
/// Frais plateforme Mastercota (organisateur) — plus de % contributeur.
class FeeQuote {
  final int net;
  final int gross;
  final int fee;

  const FeeQuote({
    required this.net,
    required this.gross,
    required this.fee,
  });
}

class PlatformPricing {
  PlatformPricing._();

  static const int freeDurationDays = 35;
  static const int maxDurationDays = 60;
  static const int extensionDays = 10;
  static const int extensionFee = 2000;

  static int? feeForDuration(int days) {
    if (days < 1 || days > maxDurationDays) return null;
    if (days <= 15) return 2000;
    if (days <= 35) return 4000;
    return 10000;
  }
}

class Fees {
  Fees._();

  static const double platformRate = 0;
  static const double paystackMomoRate = 0;
  static const double serviceRate = 0;
  static const String serviceFeeLabel = '0 %';

  static int roundUpTo5(double n) => (n / 5).ceil() * 5;

  /// Plus de surcoût : le contributeur paie le montant exact.
  static FeeQuote? fromNet(num netInput) {
    final net = netInput.round();
    if (net <= 0) return null;
    return FeeQuote(net: net, gross: net, fee: 0);
  }
}

File: frontend/lib/fees.ts
/** Frais Mastercota + Paystack MoMo CI — aligné sur backend/src/lib/fees.ts */
export const PLATFORM_RATE = 0.01;
export const PAYSTACK_MOMO_RATE = 0.0195;
export const SERVICE_RATE = PLATFORM_RATE + PAYSTACK_MOMO_RATE;
/** Libellé UI (~3 %) */
export const SERVICE_FEE_LABEL = "~3 %";

export type FeeQuote = {
  net: number;
  gross: number;
  fee: number;
};

export function roundUpTo5(n: number): number {
  return Math.ceil(n / 5) * 5;
}

export function feesFromNet(netInput: number): FeeQuote | null {
  const net = Math.round(Number(netInput));
  if (!Number.isFinite(net) || net <= 0) return null;
  const gross = roundUpTo5(net / (1 - SERVICE_RATE));
  return { net, gross, fee: gross - net };
}
